Sources/WendyAgent/Services/BluetoothManager.swift

```diff
@@ -0,0 +1,295 @@
+import Foundation
+import Logging
+import Subprocess
+
+/// Manages Bluetooth operations using BlueZ (bluetoothctl)
+struct BluetoothManager: Sendable {
+    let logger: Logger
+    private static let defaultScanTimeoutSeconds: UInt32 = 20
+
+    struct BluetoothDeviceInfo: Sendable {
+        let name: String
+        let address: String
+        let rssi: Int?
+        let paired: Bool
+        let connected: Bool
+        let trusted: Bool
+        let deviceType: String
+        let icon: String?
+    }
+
+    /// Lists Bluetooth devices using bluetoothctl
+    func listDevices(pairedOnly: Bool) async throws -> [BluetoothDeviceInfo] {
+        // First, get the list of device addresses
+        let devicesOutput = try await runBluetoothctl(["devices"])
+        let deviceAddresses = Self.parseDeviceAddresses(from: devicesOutput)
+
+        var devices: [BluetoothDeviceInfo] = []
+
+        // Get detailed info for each device
+        for address in deviceAddresses {
+            if let deviceInfo = try await getDeviceInfo(address: address) {
+                if pairedOnly && !deviceInfo.paired {
+                    continue
+                }
+                devices.append(deviceInfo)
+            }
+        }
+
+        return devices
+    }
+
+    /// Gets detailed information about a specific device
+    private func getDeviceInfo(address: String) async throws -> BluetoothDeviceInfo? {
+        let infoOutput = try await runBluetoothctl(["info", address])
+        return Self.parseDeviceInfo(from: infoOutput, address: address)
+    }
+
+    /// Starts Bluetooth discovery scan
+    func startScan(timeoutSeconds: UInt32) async throws {
+        // Power on the adapter first
+        try await powerOn()
+
+        // bluetoothctl scan needs a timeout in non-interactive mode to actually discover devices.
+        let effectiveTimeoutSeconds =
+            timeoutSeconds == 0 ? Self.defaultScanTimeoutSeconds : timeoutSeconds
+
+        _ = Task.detached { [logger, self] in
```

Comment:
This doesn't need to be a (detached) background task. Can we just call this inline?